synphot/wrapper.py

```diff
@@ -0,0 +1,153 @@
+    quantum_efficiency : list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values
+        A model for the quantum efficiency of the instrument's CCD. 
+        Default is None.
+    atmospheric_transmission : list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values
+        A model for the atmopsheric transmission.
+    """
+    @u.quantity_input(diameter=u.m, distance=u.pc, exptime=u.s)
+    def __init__(self, source_spec, diameter, distance=0 * u.pc,
+                 bandpass=SpectralElement(Const1D, amplitude=1),
+                 quantum_efficiency=None, atmospheric_transmission=None,
+                 force=None, exptime=1 * u.s):
+        self.source_spec = source_spec
+        self.diameter = diameter
+        self.aperture_area = np.pi * (diameter / 2) ** 2
+        self.distance = distance
+        self._bandpass = bandpass
+        self._qe = quantum_efficiency
+        self._at = atmospheric_transmission
+        self.exptime = exptime
+
+        # turn spectrum elements into synphot objects
+        self._set_bandpass()
+        if self._qe is not None:
+            self._include_qe()
+        if self._at is not None:
+            self._include_atmosphere()
+                    
+        self.obs = Observation(self.source_spec, self.throughput, force=force)
+        self.countrate = self.obs.countrate(area=self.aperture_area)
+        self.snr = ccd_snr(self.countrate * self.exptime)
+
+    def _set_bandpass(self):
+
+        if type(self._bandpass) == list:
+            if len(self._bandpass) != 2:
+                raise exceptions.BandpassError("bandpass should be "
+                                               "a list with 2 x n "
+                                               "dimensions.")
+            # order of waves/transmittance given by bandpass doesn't matter:
+            try:
+                self.bandpass = _call_SpecElement_4list(self._bandpass[0],
+                                                        self._bandpass[1])
+            except u.UnitsError:
+                try:
+                    self.bandpass = _call_SpecElement_4list(self._bandpass[1],
+                                                            self._bandpass[0])
+                except u.UnitsError:
+                    print("rows of bandpass list must be of type "
+                          "astropy.units.quantity.Quantity")
+                
+        elif type(self._bandpass) == str:
+            self.bandpass = SpectralElement.from_filter(self._bandpass)
+
+        elif type(self._bandpass) == SpectralElement:
+            self.bandpass = self._bandpass
+
+        else:
+            raise exceptions.BandpassError("bandpass must be of type str, "
+                "`~synphot.spectrum.SpectralElement`, or "
+                "list of shape(2, n) containing "
+                "`~astropy.units.quantity.Quantity` values")
+
+        self.throughput = self.bandpass
```

Comment:
I might be missing something here, but is there a point of having a separate `throughput` and `bandpass` object carried around if one becomes the other eventually?

Reply:
This comes from me wanting the user to be able to call `self.bandpass` to inspect the filter, while `self.throughput` would allow them to see all of the throughput effects in play (QE, atmosphere, etc).